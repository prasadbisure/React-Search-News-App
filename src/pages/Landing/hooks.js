import { useState, useEffect } from 'react';
import { useHistory } from 'react-router-dom';
import { fetchSearchKeywords } from '../../services';

const useHooks = () => {
    const [searchValue, setSearchValue] = useState('');
    const [searchResult, setSearchResult] = useState([]);
    let history = useHistory();

    const handleInputChange = (e) => {
        setSearchValue(e.target.value);       
    };

    useEffect(()=> {
        let delay = setTimeout(()=> {
                fetchSearchKeywords(searchValue).then(res=>{
                    setSearchResult(res?.data?.response?.results)
                })
            
        }, 1000)
        return () => clearTimeout(delay);
    },[searchResult, searchValue])
    
    const onHandleClick = () => {
        history.push({
            pathname: '/search',
            search: `?query=${searchValue}`,
        });   
      };

    const handleKeywordClick = (param) => () => {
        setSearchValue(param);
        history.push({
            pathname: '/search',
            search: `?query=${param}`,
        }); 
    }

    return {
        searchValue,
        searchResult: Array.from(searchResult),
        handleInputChange,
        onHandleClick,
        handleKeywordClick
    }
}
export default useHooks;