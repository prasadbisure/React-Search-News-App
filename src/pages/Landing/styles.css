h1{
    font-size: 68px;
}