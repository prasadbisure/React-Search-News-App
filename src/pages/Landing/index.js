import SearchInput from '../../components/SearchInput';
import Button from '../../components/Button';
import SearchKeyword from '../SearchKeyword'
import useHooks from './hooks';
import './styles.css';

const Landing = () => {
    const { searchValue ,searchResult, handleInputChange, onHandleClick, handleKeywordClick } = useHooks();
    return (
        <div>
            <h1>News Lister</h1>
            <SearchInput handleInputChange={handleInputChange} />
            {searchValue && (<SearchKeyword data={searchResult} handleKeywordClick={handleKeywordClick} />)}
            <Button onHandleClick={onHandleClick}>{'Search'}</Button>
        </div>
    )
};

export default Landing;
