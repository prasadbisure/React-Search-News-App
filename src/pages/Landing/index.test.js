import { shallow } from 'enzyme';
import Landing from './';

jest.mock('./hooks', () => ({
    __esModule: true,
    default: () => ({
        searchValue: 'Test value',
        searchResult: [], 
        handleInputChange: jest.fn(), 
        onHandleClick: jest.fn(), 
        handleKeywordClick: jest.fn()
    })
}));

describe('Snapshot for Landing component', () => {
    const wrapper = shallow(<Landing />);
    it('should render snapshot with default props', () => {
        expect(wrapper).toMatchSnapshot();    
    });
});
