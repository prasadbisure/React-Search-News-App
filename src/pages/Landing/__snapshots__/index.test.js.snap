// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot for Landing component should render snapshot with default props 1`] = `
<div>
  <h1>
    News Lister
  </h1>
  <ForwardRef(SearchInput)
    handleInputChange={[MockFunction]}
  />
  <SearchKeyword
    data={Array []}
    handleKeywordClick={[MockFunction]}
  />
  <Button
    onHandleClick={[MockFunction]}
  >
    Search
  </Button>
</div>
`;
